dss_datamover/master_application.py

```diff
@@ -204,7 +204,11 @@ def start(self):
         if not self.standalone:
             if not (self.operation.upper() == "LIST" and not self.config.get("distributed", False)):
                 self.stop_client_stale_process_on_clients()
-                self.spawn_clients()
+                if not self.spawn_clients():
```

Comment:
this block executes if no clients were spun up. What about the case where only a fraction of the clients did not spawn? We want to be able to handle that scenario as well. We can compare the length of the successfully spawned clients with the # of expected clients. Then we can add logic to ensure that the application is not hanging waiting for those unspawned clients.

Reply:
Response From Author: "- I have tested this scenario, as I wrote in the validation part. The application did not hang itself to wait for unspawned clients, it only got stuck when all the clients failed in spawning. The application can function normally even if only one of the clients gets spawned successfully, so I only terminate it when no clients are spawned."